mindee/cli.py

```diff
@@ -145,45 +216,50 @@ def _parse_args() -> Namespace:
         dest="product_name",
         required=True,
     )
+
     for name, info in DOCUMENTS.items():
         subp = subparsers.add_parser(name, help=info.help)
-        if name == "custom":
-            subp.add_argument(
-                "-u",
-                "--user",
-                dest="username",
-                required=True,
-                help="API account name for the endpoint",
-            )
-            subp.add_argument(
-                "-k",
-                "--key",
-                dest="api_key",
-                help="API key for the account",
+        parsers_instruction_type = subp.add_subparsers(
+            dest="instruction_type", required=True
+        )
+
+        if info.is_sync:
+            subp_predict = parsers_instruction_type.add_parser(
+                "parse", help=f"Parse {name}"
             )
-            subp.add_argument(
-                "-v",
-                "--version",
-                default="1",
-                dest="api_version",
-                help="Version for the endpoint. If not set, use the latest version of the model.",
+            _add_options(subp_predict, "predict", name)
+            subp_predict.add_argument(dest="path", help="Full path to the file")
+
+        if info.is_async:
+            parser_enqueue = parsers_instruction_type.add_parser(
+                "enqueue", help=f"Enqueue {name}"
             )
-            subp.add_argument(dest="api_name", help="Name of the API")
-        else:
-            subp.add_argument(
-                "-k",
-                "--key",
-                dest="api_key",
-                help="API key for the account",
+            _add_options(parser_enqueue, "enqueue", name)
+            parser_enqueue.add_argument(dest="path", help="Full path to the file")
+
+            parser_parse_queued = parsers_instruction_type.add_parser(
+                "parse-queued", help=f"Parse (queued) {name}"
             )
-            subp.add_argument(
-                "-w",
-                "--with-words",
-                dest="include_words",
-                action="store_true",
-                help="Include words in response",
+            _add_options(parser_parse_queued, "parse-queued", name)
+            parser_parse_queued.add_argument(
+                dest="queue_id", help="Async queue ID for a document (required)"
             )
-        subp.add_argument(
+
+    parsed_args = parser.parse_args()
+    return parsed_args
+
+
+def _add_options(parser: ArgumentParser, category: str, name: str):
```

Comment:
I'm not a big fan of this type of function, where a "magic" string argument completely changes the behavior.

Better to split into:
* `_add_sending_options` equivalent to `["predict", "enqueue"]`
* `_add_custom_options` equivalent to `name == "custom"`
* `_add_display_options` equivalent to `["predict", "parse-queued"]`

... and call the appropriate function in the `_parse_args` function.

The should make the code easier to reason about.